components/autocomplete_selector.tsx

```diff
@@ -1,29 +1,49 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import SuggestionBox from 'components/suggestion/suggestion_box';
 import SuggestionList from 'components/suggestion/suggestion_list';
 
-export default class AutocompleteSelector extends React.PureComponent {
-    static propTypes = {
-        providers: PropTypes.array.isRequired,
-        value: PropTypes.string.isRequired,
-        onSelected: PropTypes.func,
-        label: PropTypes.node,
-        labelClassName: PropTypes.string,
-        inputClassName: PropTypes.string,
-        helpText: PropTypes.node,
-        placeholder: PropTypes.string,
-        footer: PropTypes.node,
-        disabled: PropTypes.bool,
-        toggleFocus: PropTypes.func,
-        listComponent: PropTypes.elementType,
-        listPosition: PropTypes.string,
-    };
+import ModalSuggestionList from './suggestion/modal_suggestion_list';
+import Provider from './suggestion/provider';
+
+type Selected = {
```

Comment:
`Selected` type is already defined at `dialog_element.tsx`. In order to remove duplicated code, I suggest to export from here and re-use it in `dialog_element.tsx`

Reply:
@babinderrathi Thanks for the changes.

I see that you deleted `Selected` type from `autocomplete_selector` and import it from `dialog_element`. I was aiming to do the other way around: i.e. keep the `Selected` type in `autocomplete_selector` and export it from `autocomplete_selector`. Then import in `dialog_element` to re-use it. 

This is important to prevent any circular imports. Because `AutocompleteSelector` is already imported in `dialog_element`. With my suggestion above, in addition to `AutocompleteSelector`, you will also import `Selected` in `dialog_element`. 

Sorry for the double work.